### PR TITLE
Fix MiniMagick::Image.open to accepts a non-ascii character filename

### PR DESCRIPTION
**Pull request by meganemura**
From this commit (4cd5081e58810d3394d27a67219e8e4e0445d851), we use URI() for any `path_or_url`, but it raises an error for the string which contains a non-ascii character.

step to reproduce:
```ruby
> MiniMagick::Image.open("スクリーンショット 2019-07-13 1.32.01.png")
URI::InvalidURIError: URI must be ascii only "\u30B9\u30AF\u30EA\u30FC\u30F3\u30B7\u30E7\u30C3\u30C8 2019-07-13 1.32.01.png"
from /Users/meganemura/.rbenv/versions/2.6.3/lib/ruby/2.6.0/uri/rfc3986_parser.rb:21:in `split'
```

This PR fixes this.

### CODE REVIEW

File: spec/support/helpers.rb [hunk 1]
```diff
@@ -18,6 +18,7 @@ def image_path(type = :default)
         when :get_pixels          then "get_pixels.png"
         when :rgb                 then "rgb.png"
         when :rgb_tmp             then "rgb_tmp.png"
+        when :non_ascii_filename  then "スクリーンショット 2019-07-13 1.32.01.png"
```
Comment (meganemura) [inline spec/support/helpers.rb]: NOTE: `スクリーンショット` means `screenshot` in japanese.

**Comment by meganemura** [comment]
@janko Would you please review this?

File: lib/mini_magick/image.rb [hunk 2]
```diff
@@ -82,15 +82,20 @@ def self.import_pixels(blob, columns, rows, depth, map, format = 'png')
...
+      end
 
-      ext ||= File.extname(uri.path)
       ext.sub!(/:.*/, '') # hack for filenames or URLs that include a colon
 
       if uri.is_a?(URI::HTTP) || uri.is_a?(URI::FTP)
         uri.open(options) { |file| read(file, ext) }
       else
-        File.open(uri.to_s, "rb", options) { |file| read(file, ext) }
+        File.open(path_or_url, "rb", options) { |file| read(file, ext) }
```
Comment (phoet) [inline lib/mini_magick/image.rb]: just asking myself if there are useful options to pass when opening in binary mode or if instead `File.read_binary` would be sufficient.

**Comment by makmic** [comment]
Please consider looking into this issue soon, as many will run into the same issue when upgrading to 4.9.4 after the CVE notice. (@janko)

**Comment by janko** [comment]
Thanks for the patch! I'll make minor modifications, replacing the exception flow with the [logic that `Kernel#open` had](https://github.com/ruby/ruby/blob/78bc6cd8dff9ee584a447251490c88ee035d218d/lib/open-uri.rb#L15-L16).

**Comment by meganemura** [comment]
@janko 
Thanks for merging.
Could you please consider publishing patch-level release to rubygems?

**Comment by janko** [comment]
Done, 4.9.5 is released.

**Pull request merged**